### PR TITLE
markupify nl2br

### PR DESCRIPTION
**Pull request by xmo-odoo**
Currently nl2br is a pretty odd duck as it will try to work text -> text and markup -> markup, requiring explicit escaping of the input in most cases (and those that don't do that are... dodgy), as usually the point would be to convert regular text into HTML content (as a very cut down version of a lightweight markup).

This PR updates nl2br to unconditionally escape the input then convert the newlines to line breaks. It also updates nl2br_enclosure to do better.

### CODE REVIEW

File: odoo/addons/base/models/ir_qweb_fields.py [hunk 1]
```diff
@@ -35,8 +35,10 @@ def nl2br_enclose(string, enclosure_tag='div'):
     """ Like nl2br, but returns enclosed Markup allowing to better manipulate
     trusted and untrusted content. New lines added by use are trusted, other
     content is escaped. """
-    converted = nl2br(escape(string))
-    return Markup(f'<{enclosure_tag}>{converted}</{enclosure_tag}>')
+    return Markup('<{enclosure_tag}>{converted}</{enclosure_tag}>').format(
```
Comment (ryv-odoo) [inline odoo/addons/base/models/ir_qweb_fields.py:38]: Why this commit ? the diff seems useless to me (we don't really care about false positive of semgrep, right ?), also I rather prefer a f-string than the ugly format())

File: addons/l10n_it_edi/models/account_move.py [hunk 2]
```diff
@@ -42,7 +41,7 @@ def get_datetime(tree, xpath):
     """ Datetimes in FatturaPA are ISO 8601 date format, pattern '[-]CCYY-MM-DDThh:mm:ss[Z|(+|-)hh:mm]'
         Python 3.7 -> 3.11 doesn't support 'Z'.
     """
-    if (datetime_str := get_text(tree, xpath)):
+    if datetime_str := get_text(tree, xpath):
```
Comment (ryv-odoo) [inline addons/l10n_it_edi/models/account_move.py:44]: ?

**Review by ryv-odoo — APPROVED** [review]
It seems a good idea.
(I find some usage in the internal repo, maybe you should tell them about this)

File: odoo/addons/base/models/ir_qweb_fields.py (hunk 1 above)
Comment (rco-odoo) [reply odoo/addons/base/models/ir_qweb_fields.py:38]: It doesn't hurt and shows good practice IMO.

**Review by rco-odoo — APPROVED** [review]

File: odoo/addons/base/models/ir_qweb_fields.py (hunk 1 above)
Comment (ryv-odoo) [reply odoo/addons/base/models/ir_qweb_fields.py:38]: My bad

Comment (tde-banana-odoo) [reply odoo/addons/base/models/ir_qweb_fields.py:38]: Stupid people would probably gladly have an explanation of what "Avoids false positive / review site for semgrep" means ^^

**Pull request closed without merging**